### PR TITLE
`[feat]`: deprecate hardcoded configs in place of api `[DRIP-444]`

### PR DESCRIPTION
**Pull request by dcaf-mocha**
Testing done in https://github.com/dcaf-labs/drip-frontend/pull/58 with `yarn start-devnet-staging`

### CODE REVIEW

File: src/drip-client/index.ts [hunk 1]
```diff
@@ -1,51 +1,59 @@
...
   public readonly querier: DripQuerier;
   public readonly admin: DripAdmin;
+  public readonly config: DripConfig;
 
   public constructor(
     public readonly network: Network,
     public readonly provider: AnchorProvider,
-    public readonly config: Config
+    public readonly programId: Address,
+    configUrlParams:
```
Comment (dcaf-mocha) [inline src/drip-client/index.ts:21]: @dcaf-matcha is there a better way?

File: src/implementations/drip-config.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      return 'https://api.drip.dcaf.so';
+    case Network.Devnet:
+      switch (clientEnv) {
+        case ClientEnv.Production:
+          return 'https://devnet.api.drip.dcaf.so';
+        case ClientEnv.Staging:
+          return 'https://drip-backend-devnet-staging.herokuapp.com';
+      }
+  }
+  return 'localhost';
```
Comment (dcaf-matcha) [inline src/implementations/drip-config.ts:19]: Let's inject this URL in somehow

File: src/implementations/drip-config.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+export class DripConfigImpl implements DripConfig {
+  private readonly DripApi: DefaultApi;
```
Comment (dcaf-matcha) [inline src/implementations/drip-config.ts:23]: lower case bro, this isn't golang brooo

**Review by dcaf-matcha — CHANGES_REQUESTED** [review]

File: src/implementations/drip-config.ts (hunk 2 above)
Comment (dcaf-mocha) [reply src/implementations/drip-config.ts:19]: Theres two ways to initialize it, one using network and client env which will derive the url as shown here, but there is an alternative to initialize via a `url` directly. `new DripConfigImpl(url)`

File: src/implementations/drip-config.ts (hunk 3 above)
Comment (dcaf-mocha) [reply src/implementations/drip-config.ts]: > this isn't golang brooo


The real issue 😞

**Pull request merged**